src/index.js

```diff
@@ -73,7 +73,8 @@ function renderToString(vnode, context, opts, inner, isSvgMode) {
 				c._disable = c.__x = true;
 				c.props = props;
 				c.context = context;
-				if (c.componentWillMount) c.componentWillMount();
+				if (c.constructor.getDerivedStateFromProps) c.state = assign(assign({}, c.state), c.constructor.getDerivedStateFromProps(c.props, c.state));
```

Comment:
`c.constructor` here could be `nodeName`.

Reply:
@developit done
> Build "preactRenderToString" to dist:
>       1.44 kB: index.mjs
>       1.49 kB: index.js